core/pkg/ingress/annotations/ipwhitelist/main.go

```diff
@@ -56,8 +56,9 @@ func (a ipwhitelist) Parse(ing *extensions.Ingress) (interface{}, error) {
 	sort.Strings(defBackend.WhitelistSourceRange)
 
 	val, err := parser.GetStringAnnotation(whitelist, ing)
-	if err != nil {
-		return &SourceRange{CIDR: defBackend.WhitelistSourceRange}, err
+	// A missing annotation is not a problem, just use the default
+	if err == ing_errors.ErrMissingAnnotations {
+		return &SourceRange{CIDR: defBackend.WhitelistSourceRange}, nil
```

Comment:
@ashb I was wrong, the issue is related to https://github.com/kubernetes/ingress/commit/8552351af0d5b919fd581c978c1c211b02006443 #527 

@jcmoraisjr could you help me?

Why are you doing this if we already mergeLocationAnnotations on https://github.com/kubernetes/ingress/blob/8552351af0d5b919fd581c978c1c211b02006443/core/pkg/ingress/controller/controller.go#L667:L679

```
// Add annotations to location of default backend (root context)
			if len(server.Locations) > 0 {
				loc := server.Locations[0]
				mergeLocationAnnotations(loc, anns)
			}

```

This is breaking #558